joplin/base/models/__init__.py

```diff
@@ -0,0 +1,165 @@
+from django.db import models
+import os
+import graphene
+
+from modelcluster.fields import ParentalKey
+from modelcluster.models import ClusterableModel
+
+from wagtail.utils.decorators import cached_classmethod
+from wagtail.admin.edit_handlers import FieldPanel, InlinePanel, MultiFieldPanel, ObjectList, StreamFieldPanel, TabbedInterface, HelpPanel
+from wagtail.core.blocks import TextBlock, RichTextBlock, ListBlock, StreamBlock, StructBlock, URLBlock, PageChooserBlock, CharBlock
+from wagtail.core.fields import StreamField, RichTextField
+from wagtail.core.models import Page, Orderable
+from wagtail.snippets.edit_handlers import SnippetChooserPanel
+from wagtail.images.edit_handlers import ImageChooserPanel
+from wagtail.images.models import Image, AbstractImage, AbstractRendition
+from wagtail.images.blocks import ImageChooserBlock
+from wagtail.snippets.models import register_snippet
+from wagtail.search import index
+from wagtail.admin.edit_handlers import PageChooserPanel
+
+from base import blocks as custom_blocks
+from base import forms as custom_forms
+
+from .translated_image import TranslatedImage
+from .contact import Contact, ContactDayAndDuration
+from .day_and_duration import DayAndDuration
+from .location import Location
+from .map import Map
+
+from .janis_page import JanisPage, JanisBasePage
+from .home_page import HomePage
+from .theme import Theme
+from .topic_collection_page import TopicCollectionPage
+from .topic_page import TopicPage, TopicPageTopicCollection
+from .service_page import ServicePage, ServicePageTopic, ServicePageContact, ServicePageRelatedDepartments
+from .information_page import InformationPage, InformationPageRelatedDepartments, InformationPageTopic, InformationPageContact
+from .department_page import DepartmentPage, DepartmentPageDirector, DepartmentPageContact
+from .official_documents_page import OfficialDocumentPage, OfficialDocumentPageOfficialDocument, OfficialDocumentPageRelatedDepartments, OfficialDocumentPageTopic
+from .guide_page import GuidePage, GuidePageTopic, GuidePageRelatedDepartments, GuidePageContact
+
+WYSIWYG_GENERAL = ['h1', 'h2', 'h3', 'h4', 'bold', 'link', 'ul', 'ol', 'code']
+DEFAULT_MAX_LENGTH = 255
+SHORT_DESCRIPTION_LENGTH = 300
+
+
+# TODO: Remove everything below this comment
+
+class TopicCollectionPageTopicCollection(ClusterableModel):
```

Comment:
I'm not sure if we actually wanted the ability to have a topic collection as a parent of another topic collection. @desigonz I know we made some design/content choices around this before. Any clarity would be great.